scheduler/planner.go

```diff
@@ -195,78 +196,93 @@ type unitInfo struct {
 	ContainsStepbackTask bool `json:"contains_stepback_task"`
 }
 
-func (u *unitInfo) value() int64 {
-	var value int64
-
-	length := int64(len(u.TaskIDs))
-	priority := 1 + (u.TotalPriority / length)
-
-	if !u.ContainsNonGroupTasks {
-		// if all tasks in the unit are in a task group then
-		// we should give it a little bump, so that task
-		// groups tasks are sorted together even when they
-		// would also be scheduled in a version.
-		priority += length
-	}
-	if u.ContainsGenerateTask {
-		// give generators a boost so people don't have to wait twice.
-		priority = priority * u.Settings.GetGenerateTaskFactor()
-	}
+// value computes a full SortingValueBreakdown, containing the final value by which the unit
+// will be sorted in its queue (stored in TotalValue), and the numerical influences that
+// the unit's properties had on computing that final value. Currently, the formula for
+// computing this value is (custom_priority * custom_rankValue) + unit_length, where custom_priority
+// and custom_rankValue are both derived from specific properties of the unit and various
+// scheduler constants.
+func (u *unitInfo) value() task.SortingValueBreakdown {
```

Comment:
I'm fine with this change but just for my understanding, doesn't this change how the value computation works? It seems like we add length throughout in the original implementation, and here we add it at the end

Reply:
I'm not sure if I fully understand but the role length plays before and after this PR should be unchanged. Before, length is added to the original priority [here](https://github.com/evergreen-ci/evergreen/pull/8485/files#diff-7225717481fd747283fb3277021a4d0c21dfabcee92d91a3f2f25f771c6a3360L209), and is part of the various rank value equations (and is also added at the end) [here](https://github.com/evergreen-ci/evergreen/pull/8485/files#diff-7225717481fd747283fb3277021a4d0c21dfabcee92d91a3f2f25f771c6a3360L223-L267).

In these new changes, length plays the same two roles, [here](https://github.com/evergreen-ci/evergreen/pull/8485/files#diff-7225717481fd747283fb3277021a4d0c21dfabcee92d91a3f2f25f771c6a3360R262) and [here](https://github.com/evergreen-ci/evergreen/pull/8485/files#diff-7225717481fd747283fb3277021a4d0c21dfabcee92d91a3f2f25f771c6a3360R221-R245), respectively.